oak_services/src/http/mod.rs

```diff
@@ -73,30 +73,32 @@ impl From<http::header::HeaderMap<http::header::HeaderValue>> for HeaderMap {
     }
 }
 
-impl HeaderMap {
-    /// Flatten the headers into an iterator of tuples (String, http::header::HeaderValue).
-    pub fn iter(
-        &self,
-    ) -> impl Iterator<Item = (http::header::HeaderName, http::header::HeaderValue)> + '_ {
-        self.headers
-            .iter()
+impl std::iter::IntoIterator for HeaderMap {
+    type Item = (http::header::HeaderName, http::header::HeaderValue);
+    type IntoIter = Box<dyn Iterator<Item = (http::header::HeaderName, http::header::HeaderValue)>>;
```

Comment:
Why is the boxing and dynamic dispatch necessary? Is there no static type that works here?

e.g. https://doc.rust-lang.org/std/slice/struct.Iter.html

Or even better, does the "impl trait" syntax work here?

Reply:
The actual type is `std::iter::FlatMap` with some anonymous type parameters. With `impl Trait` there is the ``impl Trait` in type aliases is unstable` error message that can be fixed by adding `#![feature(type_alias_impl_trait)]`. Should I use `impl Trait`?

Reply:
Interesting, I didn't know about that feature. I don't think it's worth using it, but `FlatMap` seems a good type for this iterator, not sure about the anonymous type parameters though, do you mean they are unnameable types?

Reply:
This is the exact type: `std::iter::FlatMap<std::collections::hash_map::IntoIter<std::string::String, proto::oak::encap::HeaderValue>, std::iter::FilterMap<std::vec::IntoIter<std::vec::Vec<u8>>, [closure@/opt/my-project/oak_services/src/http/mod.rs:85:53: 100:18 name:_]>, [closure@/opt/my-project/oak_services/src/http/mod.rs:84:23: 101:14]>`. The `FilterMap` in it makes it complicated.

Reply:
If that's the case, it seems like an unnameable (not anonymous) type indeed (see https://internals.rust-lang.org/t/idea-making-unnameable-types-nameable/7797 ) -- more precisely, the type of the closure is unnameable, and it ends up being part of the reified `FlatMap` type.

Reply:
Yes. That is the problem. I did not know about the term `unnameable`. Thanks.